sklearn/isotonic.py

```diff
@@ -252,7 +278,7 @@ def _build_f(self, X, y):
             # single y, constant prediction
             self.f_ = lambda x: y.repeat(x.shape)
         else:
-            self.f_ = interpolate.interp1d(X, y, kind='slinear',
+            self.f_ = interpolate.interp1d(X, y, kind='linear',
```

Comment:
When I changed it back to linear I wan't able to find a good answer as to why it was changed to spline interpolation.

I changed it to linear because
1. This change made predictions stay the same whether or not I removed the 'unnecessary' points.
2. It makes more intuitive sense to me.